### PR TITLE
Timeout for retrying initial ssh connection

### PR DESCRIPTION
**Pull request by joshdk**

### CODE REVIEW

**Comment by joshdk** [comment]
The motivation behind this feature is that in our Drone build we dynamically provision a VM, then run commands on it for testing ([see my comments here](https://github.com/appleboy/drone-ssh/pull/112#issuecomment-368196122)).

This VM is quickly "up" from a cloud infrastructure perspective, but it is often the case where SSHD needs an additional ~30 seconds to be functional. This PR adds an additional `PLUGIN_RETRY_TIMEOUT` value that allows a developer to specify how long to attempt to connect to the host for, retrying whenever a network error is encountered, with exponential backoff.

Basically prevents your entire build from turning red, just because your SSH daemon took a few seconds too long to start 😄 

Cheers!

**Comment by joshdk** [comment]
@appleboy do you have any thoughts on this PR? I would be happy to hear any feedback you have 😄

**Comment by appleboy** [comment]
I will take this.
